### PR TITLE
refactor(kad): rename to follow naming convention across repository

### LINKED ISSUE #4485: kad: rename to follow naming convention across repository
As tracked in https://github.com/libp2p/rust-libp2p/issues/2217, we should rename the `Kademlia` behaviour to `Behaviour` and all related symbols such that users can import it as:

```rust
use libp2p::kad;

let b = kad::Behaviour::new(kad::Config::default());
```

### PR DESCRIPTION
**Pull request by PanGan21**
## Description

Renamed the following

`kad::Kademlia` -> `kad::Behaviour`
`kad::KademliaEvent` -> `kad::Event`
`kad::KademliaBucketInserts` -> `kad::BucketInserts`
`kad::KademliaStoreInserts` -> `kad::StoreInserts`
`kad::KademliaConfig` -> `kad::Config`
`kad::KademliaCaching` -> `kad::Caching`
`kad::KademliaEvent` -> `kad::Event`
`kad::KademliaConnectionType` -> `kad::ConnectionType`
`KademliaHandler` -> `Handler`
`KademliaHandlerEvent` -> `HandlerEvent`
`KademliaProtocolConfig` -> `ProtocolConfig`
`KademliaHandlerIn` -> `HandlerIn`
`KademliaRequestId` -> `RequestId`
`KademliaHandlerQueryErr` -> `HandlerQueryErr`

Resolves: #4485

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Comment by PanGan21** [comment]
I am wondering if non exported variable names from other files like `KademliaHandler` should also be renamed or it is just client faced change?

**Comment by thomaseizinger** [comment]
> I am wondering if non exported variable names from other files like `KademliaHandler` should also be renamed or it is just client faced change?

We should rename the internal ones too but we don't have to add type-aliases for those :)

**Comment by thomaseizinger** [comment]
Have a look at the CI failures. I think the type-aliases will have to go to the top-level module to be publicly visible :)

File: protocols/kad/src/lib.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@
 //! [Identify](https://github.com/libp2p/specs/tree/master/identify) protocol might be seen as a core protocol. Rust-libp2p
 //! tries to stay as generic as possible, and does not make this assumption.
 //! This means that the Identify protocol must be manually hooked up to Kademlia through calls
```
Comment (thomaseizinger) [inline protocols/kad/src/lib.rs:28]: We need to update these docs that is no longer true. Doesn't have to be in this PR though :)

**Review by thomaseizinger — COMMENTED** [review]
This is great, thank you! To finish this off, can you please:

- Bump the patch version in `protocols/kad/Cargo.toml`
- Make the same bump in `[workspace.dependencies]` in `Cargo.toml`
- Write a changelog entry in `protocols/kad/CHANGELOG.md`

**Comment by PanGan21** [comment]
> This is great, thank you! To finish this off, can you please:
> 
> * Bump the patch version in `protocols/kad/Cargo.toml`
> * Make the same bump in `[workspace.dependencies]` in `Cargo.toml`
> * Write a changelog entry in `protocols/kad/CHANGELOG.md`

Done! Thanks for the guidance! 
I started reading/learning about Kademlia while doing this pr so I will look for related issues :)

File: protocols/kad/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+## 0.44.6
```
Comment (thomaseizinger) [inline protocols/kad/CHANGELOG.md]:
```suggestion
## 0.44.6 - unreleased
```

**Review by thomaseizinger — DISMISSED** [review]
Thanks!

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**